core/src/main/java/org/infinispan/remoting/transport/jgroups/Responses.java

```diff
@@ -58,19 +64,38 @@ public static Responses suspected(Collection<Address> dests) {
     *
     * @param results Received responses.
     */
-   public Responses(RspList<Response> results) {
-      int size = results.size();
-      this.addresses = new Address[size];
-      this.responses = new AtomicReferenceArray<>(size);
-      int i = 0;
-      for (Map.Entry<Address, Rsp<Response>> e : results.entrySet()) {
-         addresses[i] = e.getKey();
-         Rsp<Response> rsp = e.getValue();
-         // keep non-received response as null
-         if (rsp.wasReceived() || rsp.wasSuspected() || rsp.wasUnreachable()) {
-            responses.set(i, rsp);
+   public Responses(Collection<Address> dests, RspList<Response> results) {
+      if (dests != null && dests.size() < results.size()) {
```

Comment:
Yeah, the condition should have been the other way around, and that's why the fix didn't really work.